AnaProd/HH_bbWW/anaTupleDef.py

```diff
@@ -86,15 +86,86 @@ def addAllVariables(dfw, syst_name, isData, trigger_class, lepton_legs, isSignal
+        dfw.DefineAndAppend(f"b{leg_idx+1}_eta", f"HbbIsValid ? static_cast<float>(HbbCandidate->leg_p4[{leg_idx}].Eta()) : -100.f")
+        dfw.DefineAndAppend(f"b{leg_idx+1}_phi", f"HbbIsValid ? static_cast<float>(HbbCandidate->leg_p4[{leg_idx}].Phi()) : -100.f")
+        dfw.DefineAndAppend(f"b{leg_idx+1}_mass", f"HbbIsValid ? static_cast<float>(HbbCandidate->leg_p4[{leg_idx}].M()) : -100.f")
+    #save information for fatjets
+    fatjet_obs = []
+    fatjet_obs.extend(FatJetObservables)
+    if not isData:
+        dfw.Define(f"FatJet_genJet_idx", f" FindMatching(FatJet_p4[FatJet_sel],GenJetAK8_p4,0.3)")
+        fatjet_obs.extend(JetObservablesMC)
+        #if isSignal:
+        #    dfw.DefineAndAppend("genchannelId","static_cast<int>(genHttCandidate->channel())")
+    dfw.DefineAndAppend(f"SelectedFatJet_pt", f"v_ops::pt(FatJet_p4[FatJet_sel])")
+    dfw.DefineAndAppend(f"SelectedFatJet_eta", f"v_ops::eta(FatJet_p4[FatJet_sel])")
+    dfw.DefineAndAppend(f"SelectedFatJet_phi", f"v_ops::phi(FatJet_p4[FatJet_sel])")
+    dfw.DefineAndAppend(f"SelectedFatJet_mass", f"v_ops::mass(FatJet_p4[FatJet_sel])")
+    for fatjetVar in fatjet_obs:
+        if(f"FatJet_{fatjetVar}" not in dfw.df.GetColumnNames()): continue
+        dfw.DefineAndAppend(f"SelectedFatJet_{fatjetVar}", f"FatJet_{fatjetVar}[FatJet_sel]")
+    subjet_obs = []
+    subjet_obs.extend(SubJetObservables)
+    if not isData:
+        dfw.Define(f"SubJet1_genJet_idx", f" FindMatching(SubJet_p4[FatJet_subJetIdx1],SubGenJetAK8_p4,0.3)")
+        dfw.Define(f"SubJet2_genJet_idx", f" FindMatching(SubJet_p4[FatJet_subJetIdx2],SubGenJetAK8_p4,0.3)")
+        fatjet_obs.extend(SubJetObservablesMC)
+    for subJetIdx in [1,2]:
+        dfw.Define(f"SelectedFatJet_subJetIdx{subJetIdx}", f"FatJet_subJetIdx{subJetIdx}[FatJet_sel]")
+        dfw.Define(f"FatJet_SubJet{subJetIdx}_isValid", f" FatJet_subJetIdx{subJetIdx} >=0 && FatJet_subJetIdx{subJetIdx} < nSubJet")
+        dfw.DefineAndAppend(f"SelectedFatJet_SubJet{subJetIdx}_isValid", f"FatJet_SubJet{subJetIdx}_isValid[FatJet_sel]")
+        for subJetVar in subjet_obs:
+            dfw.DefineAndAppend(f"SelectedFatJet_SubJet{subJetIdx}_{subJetVar}", f"""
+                                RVecF subjet_var(SelectedFatJet_pt.size(), 0.f);
+                                for(size_t fj_idx = 0; fj_idx<SelectedFatJet_pt.size(); fj_idx++) {{
+                                    auto sj_idx = SelectedFatJet_subJetIdx{subJetIdx}.at(fj_idx);
+                                    if(sj_idx >= 0 && sj_idx < SubJet_{subJetVar}.size()){{
+                                        subjet_var[fj_idx] = SubJet_{subJetVar}.at(sj_idx);
+                                    }}
+                                }}
+                                return subjet_var;
+                                """)
 
     # save all selected reco jets
     dfw.Define("centralJet_idx", "CreateIndexes(Jet_p4[Jet_sel].size())")
```

Comment:
if we store `b1`, `b2` and `ExtraJet`, we don't need to store `centralJet`

Reply:
storeExtraJets is controlled by global parameters which is false in default. So in current setup we save 
central jets, b1, b2 and only 'nExtraJets'

Reply:
We can set it to `true` (or ignore this flag in the code). Otherwise, `extraJet` and `centralJet` are redundant and could lead to confusion.

Reply:
Alternatively, why do we need b1, b2 and `ExtraJet`? We can order centralJets by pnet score and `b1_pt` would always be `centralJet_pt.at(0)`. Perhaps more flexible.. What do you think?

Reply:
Extra jet for sure we do not need. we need only b1,b2 and central jets or central jets ordered by btaging

Reply:
I propose to keep `centraJet` sorted by the btag-score, and remove `b*` and `extraJet`. It will also be more adapted for the ongoing studies on selection efficiency, mass reconstruction, etc.

Reply:
is there anything exist i can use to sort by btag score 
i quickly tried changing this from pT to btag score but did not work (is this expected ?)
https://github.com/cms-flaf/Framework/blob/637a3c8a8d64fc19fcf11a00827915c43e027336/AnaProd/HH_bbWW/anaTupleDef.py#L98-99

    dfw.Define("centralJet_idx", "CreateIndexes(Jet_btagPNetB[Jet_sel].size())")
    dfw.Define("centralJet_idxSorted", "ReorderObjects(Jet_btagPNetB[Jet_sel], centralJet_idx)")